src/runtime_src/core/edge/sw_em/generic_pcie_hal2/shim.cxx

```diff
@@ -481,6 +482,7 @@ namespace xclcpuemhal2 {
     int result = dumpXML(header, xmlFile) ;
     if (result != 0) return result ;
 
+    mHeader = const_cast<xclBin*>(header);
```

Comment:
You are storing a reference to memory that you don't own.  Caller can delete the xclbin immediately after calling xclLoadXclbin.  

You don't need this copy.  CpuemShim has a `mCoreDevice` data member, which you can use to access the IP_LAYOUT via `mCoreDevice->get_axlf_section(IP_LAYOUT)`.

Reply:
used mCoreDevice->get_axlf_section(IP_LAYOUT) in code removed copy of it.